controller/class.tapestry-controller.php

```diff
@@ -43,56 +43,76 @@ class TapestryController
         'INVALID_CHILD_NODE' => [
             'MESSAGE'   => 'Node is not a child of the tapestry',
             'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Settings do not exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_LINK' => [
+            'MESSAGE'   => 'New Tapestry link is invalid',
+            'STATUS'    => ['status' => 400]
+        ],
+        'NODES_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Nodes exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'GROUPS_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Groups exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'LINKS_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Links exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'POST_ID_ALREADY_SET' => [
+            'MESSAGE'   => 'PostID is set when creating a new Tapestry',
+            'STATUS'    => ['status' => 500]
         ]
     ];
     private $postId;
 
     /**
      * Constructor
      */
-    public function __construct($postId = 0)
+    public function __construct($postId = null)
```

Comment:
Documentation for [wp_insert_post](https://developer.wordpress.org/reference/functions/wp_insert_post/) states that it requires postId to be 0 to create a new post. I suggest you change this to 0 and also do the same below since you may be passing this to the wp_insert_post function at some point.